core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java

```diff
@@ -45,51 +45,51 @@
      * Retrieves a logger for the passed class using the {@link LoggerFactory}.
      * @param clazz Class creating the logger.
      */
-    public ServiceLogger(Class clazz) {
+    public ClientLogger(Class clazz) {
         this(clazz.getName());
     }
 
     /**
      * Retrieves a logger for the passed class name using the {@link LoggerFactory}.
      * @param className Class name creating the logger.
      */
-    public ServiceLogger(String className) {
+    public ClientLogger(String className) {
```

Comment:
Do we want to also support a factory method just like slf4j way of instantiating a logger instance instead of constructor?

Reply:
I'm OK with either approach, but I want only one. What are your main arguments for a factory approach?

Reply:
If we have many optional parameter to pass in, a Builder approach would fit. But we just have one parameter to pass in this case. So We probably should keep as it is.

Reply:
Factory design pattern is different than the builder design pattern. Factories are beneficial for testing, etc.

Reply:
I don't have a strong opinion either way for this particular use-case. But some arguments *for* a factory method would be :
1. Consistent with how some other popular `Loggers` are instantiated - log4j, logback, java.util
2. Can provide a meaningful name for factory methods. For e.g.
`Logger.getLoggerForClass(Class clazz)`
`Logger.getLoggerForApp(String applicationName)`
`Logger.getLoggerForClassName(String className)`
`Logger.getLoggerForClassNameAndApp(String appName, String className)`
These are just made up examples and are not required here. But if we wanted to support something like that above, we won't be able to do it through constructors as there are two methods with the same method signature which takes a single `String` arg.

I am fine with either approach as well. 

Also, is this logger expected to be used just within the SDK or is it also exposed to users of SDK?

Reply:
The intention is to use this only in our client libraries. Having said that, users can use it if they want (but I don't see why they would). In some places we have constructors that take a ClientLogger - so this is why we want this to be public API.

Reply:
Makes sense, thanks for the info!

Reply:
Since this is for internal Azure library and there is no strong need as of now, I will not add factory method for now. We might add in future if we see need for it. Hope that is fine.